_modules/dotnet-path.test.js

```diff
@@ -0,0 +1,504 @@
+            const fileFirstSpy = jest.spyOn(file, "first").mockImplementation(() => mockFilePath);
+
+            // Act
+            const result1 = dotnetPath.dataProjectFilePath();
+            const result2 = dotnetPath.dataProjectFilePath();
+
+            // Assert
+            // file.first() should only be called once, and the cached variable should be returned
+            // on the second call to dataProjectFilePath()
+            expect(fileFirstSpy).toHaveBeenCalledTimes(1);
+            expect(result1).toBe(mockFilePath);
+            expect(result2).toBe(mockFilePath);
+        });
+    }); // end dataProjectFilePath
+
+    /**************************************************************************************************
+     * dataProjectFilePathOrExit
+     **************************************************************************************************/
+    describe("dataProjectFilePathOrExit", () => {
+        test("when dataProjectFilePath() returns a value, it returns that value", () => {
+            // Arrange
+            const dataProjectFilePathSpy = jest.spyOn(dotnetPath, "dataProjectFilePath").mockImplementation(() => mockFilePath);
+
+            // Act
+            const result = dotnetPath.dataProjectFilePathOrExit();
+
+            // Assert
+            expect(dataProjectFilePathSpy).toHaveBeenCalledTimes(1);
+            expect(result).toBe(mockFilePath);
+        });
+
+        test.each([
+            undefined,
+            null
+        ])("when dataProjectFilePath() returns %p, it calls shell.exit", (returnValue) => {
+            // Arrange
+            const dataProjectFilePathSpy = jest.spyOn(dotnetPath, "dataProjectFilePath").mockImplementation(() => returnValue);
+            const shellExitSpy = jest.spyOn(shell, "exit").mockImplementation();
+
+            // Act
+            dotnetPath.dataProjectFilePathOrExit();
+
+            // Assert
+            expect(dataProjectFilePathSpy).toHaveBeenCalledTimes(1);
+            expect(shellExitSpy).toHaveBeenCalledWith(1);
+        });
+    }); // end dataProjectFilePathOrExit
+
+    /**************************************************************************************************
+     * releaseDir
+     **************************************************************************************************/
+    describe("releaseDir", () => {
+        test.each([
+            undefined,
+            null
+        ])("when solutionDir() returns %p, it returns undefined", (returnValue) => {
+            // Arrange
+            const solutionDirSpy = jest.spyOn(dotnetPath, "solutionDir").mockImplementation(() => returnValue);
+
+            // Act
+            const result = dotnetPath.releaseDir();
+
+            // Assert
+            expect(solutionDirSpy).toHaveBeenCalledTimes(1);
+            expect(result).toBeUndefined();
+        });
+
+        test(`when solutionDir() returns a value, it returns an absolute path to <solutionDir>/release`, () => {
```

Comment:
Same as above, if `release` is baked into the implementation, recommend exporting it and referenced by tests instead of hardcoding here.